src/bdc/steps/gpt_summarizer.py

```diff
@@ -20,6 +21,20 @@
 log = get_logger()
 
 
+def check_df(df, required_fields):
+    """
+    Check if the given DataFrame contains all the required fields.
+
+    Args:
+        df (pandas.DataFrame): The DataFrame to be checked.
+        required_fields (dict): A dictionary of required fields, where the keys are the field names and the values are not used.
+
+    Returns:
+        bool: True if the DataFrame contains all the required fields, False otherwise.
+    """
+    return df is not None and all(column in df for column in required_fields.values())
```

Comment:
This function seems very useful for all steps, as all of them check whether the required fields are in the df usually.
Maybe we can make this the base implementation for the `verify()` function of the `Step` class. Then every step can implement this by doing:
```
def verify(self):
    super.verify()
    # step specific very logic like checking for API keys
```

Reply:
I think that it would be better to do it in refactoring related ticket. I just checked the steps and some of them do not contain `required_fields` at all and refactoring it in this ticket will take longer than changing few lines. I want to merge it this week before today 5pm. On the other hand, I totally agree with refactoring the steps and making them follow the same pattern.

Reply:
Addressed, after merging dev into current, I was able to refactor it trivially.